sdk/eventhub/event-hubs/src/eventHubProducerClient.ts

```diff
@@ -212,11 +244,52 @@ export class EventHubProducerClient {
     return new EventDataBatchImpl(
       this._context,
       maxMessageSize,
+      Boolean(this._enableIdempotentPartitions),
       options.partitionKey,
-      options.partitionId
+      partitionId
     );
   }
 
+  /**
+   * Get the information about the state of publishing for a partition as observed by the `EventHubProducerClient`.
+   * This data can always be read, but will only be populated with information relevant to the active features
+   * for the producer client.
+   *
+   * @param partitionId - Id of the partition from which to retrieve publishing properties.
+   * @param options - The set of options to apply to the operation call.
+   * - `abortSignal`  : A signal the request to cancel the send operation.
+   * @returns Promise<void>
+   * @throws AbortError if the operation is cancelled via the abortSignal.
+   */
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore this is called in EventHubBufferedProducerClient via cast-to-any workaround
+  private async getPartitionPublishingProperties(
+    partitionId: string,
+    options: OperationOptions = {}
+  ): Promise<PartitionPublishingProperties> {
+    if (!isDefined(partitionId)) {
+      throw new TypeError(
+        `getPartitionPublishingProperties called without required argument "partitionId"`
+      );
+    }
+
+    if (typeof partitionId === "number") {
+      partitionId = String(partitionId);
+    }
```

Comment:
this is weird given that the type of `partitionId` is `string`

Reply:
I've seen many runtime checks in Service Bus code. this might be one of them.

Reply:
Yeah these checks seem to be leftovers from JavaScript code. I would move/remove them as appropriate because if we do not trust our types, then it is meaningless to write them down.